managed/services/backup/compatibility_service.go

```diff
@@ -188,3 +214,36 @@ func (s *CompatibilityService) FindArtifactCompatibleServices(
 
 	return compatibleServices, nil
 }
+
+// CheckArtifactCompatibility check compatibility between artifact and target database.
+func (s *CompatibilityService) CheckArtifactCompatibility(artifactID, targetDBVersion string) error {
+	artifactModel, err := models.FindArtifactByID(s.db.Querier, artifactID)
+	if err != nil {
+		return err
+	}
+
+	serviceModel, err := models.FindServiceByID(s.db.Querier, artifactModel.ServiceID)
+	if err != nil {
+		return err
+	}
+
+	return s.artifactCompatibility(artifactModel, serviceModel, targetDBVersion)
+}
+
+// artifactCompatibility contains logic for CheckArtifactCompatibility
+func (s *CompatibilityService) artifactCompatibility(artifactModel *models.Artifact, serviceModel *models.Service, targetDBVersion string) error {
+	var err error
+	if artifactModel.DBVersion != "" && artifactModel.DBVersion != targetDBVersion {
```

Comment:
iirc `pbm` only requires that the mongo version is within the same "major version" - so this is a stricter limit and we'll have to communicate it I guess.

Reply:
You are right, but since we don't have any agreement on that it's better to restrict version to exactly the same. Whenever we have an agreement we can rely on that and on tests that should be done to confirm everything is working as expected.